### PR TITLE
 Add `enaddr` sqlite function for encoding a bitmessage address

### PR DESCRIPTION
**Pull request by cis-muzahid**
Add `enaddr` sqlite function so that this operation (encoding addresses) can be performed inside sql statements.

### CODE REVIEW

File: src/tests/test_sqlthread.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+        encoded_str = encodeAddress(4, 1, "21122112211221122112")
+
+        # Initialise Database
+        self.initialise_database("create_function")
+
+        sqlExecute('''INSERT INTO testhash (addressversion, hash) VALUES(4, "21122112211221122112")''')
+        # call function in query
+
+        sqlExecute('''UPDATE testhash SET address=(enaddr(testhash.addressversion, 1, hash)) WHERE hash=testhash.hash''')
```
Comment (PeterSurda) [inline src/tests/test_sqlthread.py:68]: why is there a `WHERE`, can't we skip it?

File: src/class_sqlThread.py [hunk 2]
```diff
@@ -358,16 +362,11 @@ def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-s
...
-            from addresses import encodeAddress
-            for row in queryResult:
-                addressHash, addressVersion = row
-                address = encodeAddress(addressVersion, 1, hash)
-                item = '''UPDATE pubkeys SET address=? WHERE hash=?;'''
-                parameters = (address, addressHash)
-                self.cur.execute(item, parameters)
+            self.cur.execute('''ALTER TABLE pubkeys ADD address text DEFAULT '' ;''')
+
+            # replica for loop to update hashed address
```
Comment (PeterSurda) [inline src/class_sqlThread.py:367]: why `WHERE`?

**Review by PeterSurda — CHANGES_REQUESTED** [review]
I think `WHERE` can be removed.

Python 3 tests fail due to `src.queues.py` not being python3 compatible.

The rest looks ok but I'll re-check once the existing problems are fixed.

**Comment by PeterSurda** [comment]
bitmessagecli can be put into a separate PR.

File: src/bmconfigparser.py [hunk 3]
```diff
@@ -89,7 +89,7 @@ def get(self, section, option, raw=False, vars=None):
            except (ConfigParser.NoSectionError, ConfigParser.NoOptionError) as e:
                 try:
                     return BMConfigDefaults[section][option]
-                except (KeyError, ValueError, AttributeError):
+                except (KeyError, ValueError, AttributeError) as e:
```
Comment (PeterSurda) [inline src/bmconfigparser.py:92]: I think the original is more correct

File: src/tests/test_sqlthread.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                sql.execute("drop table if exists %s" % q)
+
+    @classmethod
+    def tearDown(cls):
+        pass
+
+    @classmethod
+    def tearDownClass(cls):
+        # Stop sql thread
+        sqlStoredProcedure('exit')
```
Comment (PeterSurda) [inline src/tests/test_sqlthread.py:46]: you should also
`sqlLokup.join()`

Comment (PeterSurda) [reply src/tests/test_sqlthread.py:46]: or set `sqlLookup.daemon = true`, that should work instead too.

**Review by PeterSurda — CHANGES_REQUESTED** [review]
- please make the changes requested
- fix python3

File: src/bmconfigparser.py [hunk 5]
```diff
@@ -53,18 +53,14 @@
 
 @Singleton
 class BMConfigParser(SafeConfigParser):
-
-    """
-    Singleton class inherited from :class:`ConfigParser.SafeConfigParser`
-    with additional methods specific to bitmessage config.
-    """
+    """Singleton class inherited from ConfigParsedadfeConfigParser
```
Comment (PeterSurda) [inline src/bmconfigparser.py:56]: why

File: src/bmconfigparser.py [hunk 6]
```diff
@@ -145,43 +148,75 @@ def safeGet(self, section, option, default=None):
...
-                        try:
-                            newVal = BMConfigDefaults[section][option]
-                        except KeyError:
-                            continue
-                        ConfigParser.ConfigParser.set(
-                            self, section, option, newVal)
-                except ConfigParser.InterpolationError:
-                    continue
+    if sys.version_info[0] == 3:
+        def addresses(self, hidden=False):
```
Comment (PeterSurda) [inline src/bmconfigparser.py:157]: why not `@staticmethod`?

File: src/bmconfigparser.py [hunk 7]
```diff
@@ -145,43 +148,75 @@ def safeGet(self, section, option, default=None):
...
+            """Return a list of local bitmessage addresses (from section labels)"""
+            return [x for x in BMConfigParser().sections() if x.startswith('BM-') and (
+                    hidden or not BMConfigParser().safeGetBoolean(x, 'hidden'))]
+
+        def paymentaddress(self):
```
Comment (PeterSurda) [inline src/bmconfigparser.py:162]: what is this

**Review by PeterSurda — CHANGES_REQUESTED** [review]
Please respond to comments and make the requested changes.

File: src/bmconfigparser.py (hunk 5 above)
Comment (cis-muzahid) [reply src/bmconfigparser.py:56]: I will remove this

File: src/debug.py [hunk 8]
```diff
@@ -149,7 +157,13 @@ def resetLogging():
 
 
 # !
-preconfigured, msg = configureLogging()
+
 logger = logging.getLogger('default')
-if msg:
-    logger.log(logging.WARNING if preconfigured else logging.INFO, msg)
+
+try:
```
[the remark this answers is not in the document]
Comment (cis-muzahid) [reply src/debug.py:167]: Okay I will apply and test

**Review by PeterSurda — APPROVED** [review]

**Pull request merged**